### PR TITLE
EWPP-2018: Add Highlighted field for Text with featured Media paragraph.

### PR DESCRIPTION
**Pull request by 22Alexandra**

### CODE REVIEW

File: modules/oe_paragraphs_media/oe_paragraphs_media.post_update.php [hunk 1]
```diff
@@ -80,3 +80,29 @@ function oe_paragraphs_media_post_update_00003(): void {
     $display->save();
   }
 }
+
+/**
+ * Add Highlighted field to Text with featured Media paragraph.
```
Comment (sergepavle) [inline modules/oe_paragraphs_media/oe_paragraphs_media.post_update.php:85]:
```suggestion
 * Add Highlighted field to Text with Featured media paragraph.
```

File: modules/oe_paragraphs_media/oe_paragraphs_media.post_update.php [hunk 2]
```diff
@@ -80,3 +80,29 @@ function oe_paragraphs_media_post_update_00003(): void {
...
+ */
+function oe_paragraphs_media_post_update_00004() {
+  $storage = new FileStorage(drupal_get_path('module', 'oe_paragraphs_media') . '/config/post_updates/00004_highlighted_field');
+  $config_manager = \Drupal::service('config.manager');
+  $entity_type_manager = \Drupal::entityTypeManager();
+  $config = [
```
Comment (sergepavle) [inline modules/oe_paragraphs_media/oe_paragraphs_media.post_update.php:91]:
```suggestion
  $configs = [
```

File: modules/oe_paragraphs_media/oe_paragraphs_media.post_update.php [hunk 3]
```diff
@@ -80,3 +80,29 @@ function oe_paragraphs_media_post_update_00003(): void {
...
+    'field.storage.paragraph.field_oe_highlighted',
+    'field.field.paragraph.oe_text_feature_media.field_oe_highlighted',
+  ];
+  foreach ($config as $config_id) {
```
Comment (sergepavle) [inline modules/oe_paragraphs_media/oe_paragraphs_media.post_update.php:95]:
```suggestion
  foreach ($configs as $config_id) {
```

**Review by sergepavle — CHANGES_REQUESTED** [review]

**Review by sergepavle — APPROVED** [review]

File: modules/oe_paragraphs_media/config/install/field.storage.paragraph.field_oe_highlighted.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    - paragraphs
+id: paragraph.field_oe_highlighted
+field_name: field_oe_highlighted
+entity_type: paragraph
+type: boolean
+settings: {  }
+module: core
+locked: false
+cardinality: 1
+translatable: true
```
Comment (yenyasinn) [inline modules/oe_paragraphs_media/config/install/field.storage.paragraph.field_oe_highlighted.yml:14]:
```suggestion
translatable: false
```

File: modules/oe_paragraphs_media/config/post_updates/00004_highlighted_field/field.storage.paragraph.field_oe_highlighted.yml [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    - paragraphs
+id: paragraph.field_oe_highlighted
+field_name: field_oe_highlighted
+entity_type: paragraph
+type: boolean
+settings: {  }
+module: core
+locked: false
+cardinality: 1
+translatable: true
```
Comment (yenyasinn) [inline modules/oe_paragraphs_media/config/post_updates/00004_highlighted_field/field.storage.paragraph.field_oe_highlighted.yml:14]:
```suggestion
translatable: false
```

File: modules/oe_paragraphs_media/oe_paragraphs_media.post_update.php [hunk 6]
```diff
@@ -80,3 +80,29 @@ function oe_paragraphs_media_post_update_00003(): void {
     $display->save();
   }
 }
+
+/**
+ * Add Highlighted field to Text with Featured media paragraph.
+ */
+function oe_paragraphs_media_post_update_00004() {
+  $storage = new FileStorage(drupal_get_path('module', 'oe_paragraphs_media') . '/config/post_updates/00004_highlighted_field');
```
Comment (yenyasinn) [inline modules/oe_paragraphs_media/oe_paragraphs_media.post_update.php:88]:
```suggestion
  $storage = new FileStorage(\Drupal::service('extension.list.module')->getPath('oe_paragraphs_media') . '/config/post_updates/00004_highlighted_field');
```
`drupal_get_path` is deprecated.

Comment (yenyasinn) [reply modules/oe_paragraphs_media/oe_paragraphs_media.post_update.php:88]: It works in D9.2 but no need to rewrite later. It is already used in other components and ewcms

**Review by yenyasinn — DISMISSED** [review]

**Review by sergepavle — DISMISSED** [review]

**Review by sergepavle — APPROVED** [review]

**Review by upchuk — APPROVED** [review]
Blind

**Pull request merged**